### DISCUSSION TITLE
Feature request - Retract on travel.

### DISCUSSION THREAD

**Discussion by OCautomotive** [supermerill/SuperSlicer#3124] [Q&A] [answered]
I just started printing with a .6mm nozzle and I noticed there is a little blob when it crosses the perimeter of my print to the other side. The same thing happens when it comes back. I can massively reduce this with the "Avoid crossing perimeters" option, but it only avoids them; so some of these travels still exist. 
I was thinking that a retract before doing the travel would fix the issue, but there doesn't seem to be an option for that. 

You can see how its crossing the perimeters normally:
![image](https://user-images.githubusercontent.com/33736638/186525627-c9aef26a-27d1-4150-badb-545341954b4e.png)

After enabling "avoid crossing perimeters" and setting "max detour" to 100% it looks like this:
![image](https://user-images.githubusercontent.com/33736638/186525904-de76ae41-687f-4020-9c03-634e8d6d5991.png)
much better, but not perfect.

I think if there was an option to retract on travel, I wouldnt even need the "Avoid crossing perimeters" option. 
It might be nice to put a option for "Min travel distance before retract" so that short travels dont trigger the retract.

**Comment by OCautomotive** [comment]
I'm a little confused. It looks like it already does retracts on these moves. Not sure when though.

**Comment by neophyl** [comment, marked as answer]
There are many retraction settings under the extruder settings  (Printer Settings>Extruder)  Retraction before travel is built in and a normal part of pretty much most slicers.  One of the settings under extruder is the minimum travel after retraction that stops it retracting if the travel isnt very far.
